.github/workflows/check_changelog.yml

```diff
@@ -0,0 +1,33 @@
+name: Check Changelog
+# This check makes sure that the changelog is properly updated
+# when a PR introduces a change in a test file.
+# To bypass this check, label the PR with  no-changelog.
+on:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'no-changelog') == 0 }}
+    steps:
+      - name: Get PR number
+        run: |
+          echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Check the changelog only if tests have been modified
+        run: |
+          set -xe
+          tests=$(git diff --name-only origin/master)
+          if [[ "$tests" =~ tests ]]
+          then
+            result=$(cat ./doc/whats_new/v*.rst)
```

Comment:
If the PR is flagged for a particular milestone it would be great to check that the changelog entry is written in the matching changelog file (for the first two components of the version number).